public/sw.js

```
@@ -37,11 +37,32 @@ self.addEventListener('notificationclick', (event) => {
   const targetUrl = (event.notification.data && event.notification.data.url) || '/'
 
   event.waitUntil(
-    self.clients.matchAll({ type: 'window', includeUncontrolled: true }).then((clientList) => {
-      // Focus an existing tab for this game if one is open, otherwise open a new one.
+    self.clients.matchAll({ type: 'window', includeUncontrolled: true }).then(async (clientList) => {
+      // Prefer a tab that's already on the target URL — just focus it.
       for (const client of clientList) {
+        try {
+          const clientPath = new URL(client.url).pathname
```

ESLint

[error] 'URL' is not defined.

(no-undef)